AzureMonitorAgent/agent.py

```diff
@@ -697,7 +698,20 @@ def handle_mcs_config(public_settings, protected_settings, default_configs):
                 log_and_exit("Enable", MissingorInvalidParameterErrorCode, 'Parameter "username" and "password" not in proxy protected setting')
         else:
             set_proxy(default_configs["MDSD_PROXY_ADDRESS"], "", "")
-
+    
+    # is this Arc? If so, check for proxy     
+    if os.path.isfile(ArcSettingsFile):
+        f = open(ArcSettingsFile, "r")
+        data = f.read()
+
+        if (data != ''):
+            json_data = json.loads(data)
+            if "proxy.url" in json_data:
+                url = json_data["proxy.url"]
+                if url != '':
```

Comment:
Need error handling when json_data is null: 
if json_data is not None and "proxy.url" in json_data:

and error handling when proxy.url is specified but empty or invalid similar to line#698. If ignoring invalid url is intentional we would not need error handling but we will likely need it to be logged

Reply:
yeah, the only validation we do is to check if its non-empty, cx has to ensure config is correct